### DISCUSSION TITLE
How to connect from AppSmith to Supabase?

### DISCUSSION THREAD

**Discussion by DanielKehoe** [supabase/supabase#1876] [Questions] [answered]
I'd like to connect to Supabase from AppSmith (a no-code CRUD front-end app). 

I've tried my URL endpoint found in the Supabase admin console under Settings/API:
```
https://vsviqhljxaohkquhjolq.supabase.co/
```
with a header `Authorization` with value `Bearer <JWT Secret>`
using the JWT Secret found in the Supabase admin console under Settings/API. 

I get an error: `no Route matched with those values`.

Some questions: 
- Should I use POST or GET?
- Am I using the correct URL? Should it contain a path?
- What should my Request Body be?
- Do I need additional Headers?

Finally, where is the documentation for setting up a connection to a Supabase URL endpoint? Seems I am having trouble finding it.

![appsmith-api-settings](https://user-images.githubusercontent.com/950301/121192729-cd05d980-c89f-11eb-95f9-719e0ef54957.png)

**Comment by inian** [comment, marked as answer]
Hey Daniel, Our Postgrest API is at `https://vsviqhljxaohkquhjolq.supabase.co/rest/v1`

**Reply by soedirgo** [reply]
@DanielKehoe you might also need to pass an `apikey` from Settings > API either as a query parameter or a header.

**Reply by DanielKehoe** [reply]
What is the correct key for the header? I have key `Authorization` with value `Bearer <JWT Secret>`. What header key should I add? Can you show me an example header? Like this `anon-public`? Or what?

Also should I send POST or GET?

**Reply by soedirgo** [reply]
The REST endpoint, which inian linked above, is documented [here](https://postgrest.org/en/stable/api.html). Which HTTP verb you use will depend on what you want to do.

You'll need the `Authorization` header to use RLS, but otherwise `apikey` should be enough. The header/query parameter name is `apikey`, and the value should be one of the keys in Settings > API > API Keys.

**Reply by DanielKehoe** [reply]
Great! That's the info I needed. Thanks!
